### PR TITLE
Updated service versions and sas recordings to stg82

### PR DESCRIPTION
**Pull request by rickle-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceVersion.java [hunk 1]
```diff
@@ -62,7 +62,12 @@ public enum BlobServiceVersion implements ServiceVersion {
     /**
      * Service version {@code 2021-04-10}.
      */
-    V2021_04_10("2021-04-10");
+    V2021_04_10("2021-04-10"),
+
+    /**
+     * Service version {@code 2021-06-08}.
+     */
+    V2021_06_08("2021-06-08");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceVersion.java:70]: `getLatest` needs to be updated to this

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeServiceVersion.java [hunk 2]
```diff
@@ -62,7 +62,12 @@ public enum DataLakeServiceVersion implements ServiceVersion {
     /**
      * Service version {@code 2021-04-10}.
      */
-    V2021_04_10("2021-04-10");
+    V2021_04_10("2021-04-10"),
+
+    /**
+     * Service version {@code 2021-06-08}.
+     */
+    V2021_06_08("2021-06-08");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeServiceVersion.java:70]: Same comment

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareServiceVersion.java [hunk 3]
```diff
@@ -63,7 +63,12 @@ public enum ShareServiceVersion implements ServiceVersion {
     /**
      * Service version {@code 2021-04-10}.
      */
-    V2021_04_10("2021-04-10");
+    V2021_04_10("2021-04-10"),
+
+    /**
+     * Service version {@code 2021-06-08}.
+     */
+    V2021_06_08("2021-06-08");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareServiceVersion.java:71]: Same comment

File: sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceVersion.java [hunk 4]
```diff
@@ -62,7 +62,12 @@ public enum QueueServiceVersion implements ServiceVersion {
     /**
      * Service version {@code 2021-04-10}.
      */
-    V2021_04_10("2021-04-10");
+    V2021_04_10("2021-04-10"),
+
+    /**
+     * Service version {@code 2021-06-08}.
+     */
+    V2021_06_08("2021-06-08");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceVersion.java:70]: Same comment

**Review by alzimmermsft — APPROVED** [review]

**Comment by rickle-msft** [comment]
We don't update getLatest until the service version roll out is complete. So that change will come later.

**Pull request merged**